src/dmd/parse.d

```diff
@@ -3719,6 +3720,27 @@ final class Parser(AST) : Lexer
             t = parseVector();
             break;
 
+        case TOK.traits:
+        {
+            AST.TraitsExp te = cast(AST.TraitsExp) parsePrimaryExp();
+            if (!te)
+            {
+                // error already emitted while parsing primary
+                t = new AST.TypeError;
+            }
+            else if (te.ident != Id.getMember)
+            {
+                // even if this is not a grammar error, it's not worth continuing.
+                error("invalid `__traits`, only `getMember` can give types and symbols");
```

Comment:
1) That should not be a parsing error.
2) What about `getOverloads`, `getVirtualMethods`, `getVirtualFunctions`, `getUnittests` and `parent` ?